shell/common/shell.h

```diff
@@ -101,7 +99,9 @@ class Shell {
   TracingController tracing_controller_;
 
   std::vector<ftl::WeakPtr<Rasterizer>> rasterizers_;
-  std::vector<ftl::WeakPtr<PlatformView>> platform_views_;
+  std::vector<std::weak_ptr<PlatformView>> platform_views_;
+
+  std::recursive_mutex mutex_;
```

Comment:
WaitForPlatformViewIds can access platform_views_ directly while holding the mutex instead of calling GetPlatformViews.  This will also avoid the list copy done by GetPlatformViews.